### PR TITLE
http:server: Document the api

### PR DESCRIPTION
**Pull request by ceolin**

### CODE REVIEW

**Comment by lpereira** [comment]
Rubberstamp+.

File: src/lib/comms/include/sol-http-server.h [hunk 1]
```diff
@@ -56,25 +56,126 @@ extern "C" {
 struct sol_http_server;
 struct sol_http_request;
 
+/**
+ * It creates a http's server on that will bind on all interfaces in the given port.
```
Comment (lpereira) [inline src/lib/comms/include/sol-http-server.h]: HTTP. It's an acronym. No need to start with `It`:

   Creates an HTTP server, binding on all interfaces at the specified @port.

File: src/lib/comms/include/sol-http-server.h [hunk 2]
```diff
@@ -56,25 +56,126 @@ extern "C" {
...
+ * same port will result in fail.
+ *
+ * @param port The port where the server will bind.
+ *
+ * @return a handle to the server on success, otherwise @c NULL is returned.
+ */
 struct sol_http_server *sol_http_server_new(uint16_t port);
+
+/**
+ * Deletes the server and consequently all handlers and dir added.
```
Comment (lpereira) [inline src/lib/comms/include/sol-http-server.h]: Will it also destroy the directories I've added, removing all the files contained therein? That sounds dangerous.

File: src/lib/comms/include/sol-http-server.h [hunk 3]
```diff
@@ -56,25 +56,126 @@ extern "C" {
...
+ * return automatically a response with the code 304 (not modified) when the request contains the header
+ * If-Since-Modified greater than the value given in this function.
+ *
+ * @param server The value got with @c sol_http_server_new
+ * @param path The same path given on @c sol_http_server_register_handler
+ * @param modified The time it was modified
+ *
+ * @return '0' on success, error code (always negative) otherwise.
+ */
 int sol_http_server_set_last_modified(struct sol_http_server *server, const char *path, time_t modified);
```
Comment (lpereira) [inline src/lib/comms/include/sol-http-server.h]: Is that global per instance of the server? I don't get how this API would be useful...

File: src/lib/comms/include/sol-http-server.h (hunk 2 above)
Comment (ceolin) [inline src/lib/comms/include/sol-http-server.h]: nope, i want to say that is not need remove the added dirs

Comment (lpereira) [inline src/lib/comms/include/sol-http-server.h]: I know, but the documentation is dubious and scary-sounding. I'd just say that this destroys the server instance, leaving the removal of handlers and directory references implicit.

File: src/lib/comms/include/sol-http-server.h (hunk 3 above)
Comment (ceolin) [inline src/lib/comms/include/sol-http-server.h]: nope, it's path's specific

File: src/lib/comms/include/sol-http-server.h (hunk 2 above)
Comment (ceolin) [inline src/lib/comms/include/sol-http-server.h]: ack

**Comment by ceolin** [comment]
merged with the fixes. Thanks

File: src/lib/comms/include/sol-http-server.h (hunk 3 above)
Comment (barbieri) [inline src/lib/comms/include/sol-http-server.h]: then you should always get from `struct stat::st_mtime`

Comment (lpereira) [inline src/lib/comms/include/sol-http-server.h]: If `path` is a file, then,  yeah, it's possible. Otherwise, it doesn't make much sense.

Comment (barbieri) [inline src/lib/comms/include/sol-http-server.h]: are we able to server something that is not a file?

if we're generating index.html, then we can use the newest file as the index.html mtime.

**Pull request closed without merging**